### PR TITLE
Parsing cpp types

### PR DESCRIPTION
**Pull request by soumilbaldota**
Addded
- simple dictionary to map c++ types to Julia types
- using a eval-repr loop to test the conversions

### CODE REVIEW

**Comment by soumilbaldota** [comment]
@tmadlener In our meeting earlier 
https://github.com/soumilbaldota/PODIO_Julia_Interface_Prototype/blob/6475b130ffa0931442deb0b38d4052746421c8c2/src/parse_type.jl#L18-L20
The necessary logic should be added to the above function such that 
- int(2^n)t should translate to the corresponding Julia type
- same with uint
- also std::array<primitive_type,n> should correspond to a julia type where primitive types are {float,int} etc..

Lastly one question:
- should we be returning the julia type or the python type,  as you had mentioned in the meet the conversion between the two is trivial
- Also have I missed anything.

**Comment by tmadlener** [comment]
Just one minor addition / clarification to the requirements: The `std::array<T, N>` case should support all types that are also allowed as usual members, i.e.:

- `T` is a `builtin` type, i.e. `int`, `float`, `double`, including the fixed width integers `intN_t` and `uintN_t`
- `T` is a `component` defined in the datamodel yaml file

> - should we be returning the julia type or the python type, as you had mentioned in the meet the conversion between the two is trivial

This should return the Julia type as we will be using it to generate julia code from it. What I meant with python is that this function has to be ultimately be implemented in python, as the code generator is implemented in python.

File: test/parsed_julia_type.jl [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+@test eval(Meta.parse(get_julia_type("int16_t"))) == Int16
+@test eval(Meta.parse(get_julia_type("uint16_t"))) == UInt16
+@test eval(Meta.parse(get_julia_type("int32_t"))) == Int32
+@test eval(Meta.parse(get_julia_type("uint32_t"))) == UInt32
+@test eval(Meta.parse(get_julia_type("int64_t"))) == Int64
+@test eval(Meta.parse(get_julia_type("uint64_t"))) == UInt64
+@test eval(Meta.parse(get_julia_type("std::array<float,6>"))) == MArray{Tuple{6}, Float32, 1, 6}
+@test eval(Meta.parse(get_julia_type("std::array<float,6>"))) == MVector{6,Float32}
+@test eval(Meta.parse(get_julia_type("std::array<float,21>"))) == MArray{Tuple{21}, Float32, 1, 21}
+@test eval(Meta.parse(get_julia_type("std::array<float,21>"))) == MVector{21,Float32}
```
Comment (tmadlener) [inline test/parsed_julia_type.jl:19]:
```suggestion
@test eval(Meta.parse(get_julia_type("std::array<float,21>"))) == MVector{21,Float32}
@test eval(Meta.parse(get_juliat_type("std::array<Vector3f, 42>"))) == MVector{42, Vector3f}
@test eval(Meta.parse(get_julia_type("std::array<uint32_t, 100>"))) == MVector{100, UInt32}
```

Just adding a few more test cases that should also work.

File: test/parsed_julia_type.jl [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+@test eval(Meta.parse(get_julia_type("int"))) == Int64
+@test eval(Meta.parse(get_julia_type("float"))) == Float32
+@test eval(Meta.parse(get_julia_type("int8_t"))) == Int8
+@test eval(Meta.parse(get_julia_type("uint8_t"))) == UInt8
+@test eval(Meta.parse(get_julia_type("int16_t"))) == Int16
+@test eval(Meta.parse(get_julia_type("uint16_t"))) == UInt16
+@test eval(Meta.parse(get_julia_type("int32_t"))) == Int32
+@test eval(Meta.parse(get_julia_type("uint32_t"))) == UInt32
+@test eval(Meta.parse(get_julia_type("int64_t"))) == Int64
+@test eval(Meta.parse(get_julia_type("uint64_t"))) == UInt64
```
Comment (tmadlener) [inline test/parsed_julia_type.jl:15]:
```suggestion
@test eval(Meta.parse(get_julia_type("uint64_t"))) == UInt64
@test eval(Meta.parse(get_julia_type("Vector3f"))) == Vector3f
@test eval(Meta.parse(get_julia_type("SomeComponent"))) == SomeComponent
```

Just a quick note on these tests: Here you actually check that what Julia parses from the output of `get_julia_type` is the same as the expected type. For the podio case I am not entirely sure that will be possible for all the use cases, as some of the types that we try to compare against will only exist after we have generated the code. One way out would be to just compare the string output of `get_juliat_type`, i.e.
```julia
@test get_julia_type("Component") == "Component"
@test get_julia_type("std::array<int32_t, 3>") == "MVector{3, Int32}"
```

Comment (tmadlener) [reply test/parsed_julia_type.jl:15]: But also keep in mind that this functionality has to be moved to podio and python in any case, so don't spend too much time on the tests here, as they will also need to be moved to podio (and python).

**Comment by soumilbaldota** [comment]
@tmadlener I have made a python function of the same and made a pull request to the julia branch in podio in generator_utils.py 
https://github.com/AIDASoft/podio/pull/310

I think this is not required by the Interface. so I am currently not adding it here.

**Pull request closed without merging**